pg_shard.c

```diff
@@ -641,14 +679,13 @@ PlanSequentialScan(Query *query, int cursorOptions, ParamListInfo boundParams)
  * query. This new query can then be pushed down to the worker nodes.
  */
 static Query *
-RowAndColumnFilterQuery(Query *query)
+RowAndColumnFilterQuery(Query *query, List *remoteRestrictList, List *localRestrictList)
 {
 	Query *filterQuery = NULL;
 	List *rangeTableList = NIL;
-	List *whereClauseList = NIL;
-	List *whereClauseColumnList = NIL;
-	List *projectColumnList = NIL;
 	List *columnList = NIL;
+	List *localColumnList = NIL;
```

Comment:
They were named after `localConds` and `remoteConds` in `postgres_fdw`. I find the `local`/`remote` distinction _very_ explicit/clear, as it immediately tells me what a variable is used for (local or remote execution). `whereClause` says where we got the value, not what it is for.

But while writing this I remember wondering if the naming couldn't be better, so I want to dump all the `local`/`remote` names here and we can discuss which need changing:

  * `localQuery` — The `Query` structure used to build a local sequential scan
  * `localRestrictList` — A list of expressions from `WHERE` clauses that must be included in the local sequential scan. Currently always `NIL`.
  * `localColumnList` — The full list of columns referenced by the `localRestrictList` quals
  * `remoteRestrictList` — A list of expressions from `WHERE` clauses that will be executed remotely. Currently contains all `WHERE` clauses from the original query.

Was it just `localColumnList` you had a problem with, or are any of the others problematic? Some of these occur both as variable names and as parameter names.